src/views/ModelPlan/Status/index.tsx

```diff
@@ -29,44 +29,59 @@ type StatusFormProps = {
 
 const Status = () => {
   const { t: modelPlanT } = useTranslation('modelPlan');
+  const { t: modelPlanTaskListT } = useTranslation('modelPlanTaskList');
   const { t: modelPlanMiscT } = useTranslation('modelPlanMisc');
   const { t: miscellaneousT } = useTranslation('miscellaneous');
 
   const { status: statusConfig } = usePlanTranslation('modelPlan');
 
+  const { showMessageOnNextPage } = useMessage();
+
   const { modelID } = useParams<{ modelID: string }>();
 
   const history = useHistory();
+
+  const params = useMemo(() => {
+    return new URLSearchParams(history.location.search);
+  }, [history.location.search]);
+
+  // Get model status from generated email link
+  const modelStatus = params.get('model-status') as ModelStatus;
+
   const formikRef = useRef<FormikProps<StatusFormProps>>(null);
-  const validationSchema = Yup.object().shape({
-    status: Yup.string().required('Enter a role for this team member')
-  });
 
   const { status } = useContext(ModelInfoContext);
 
   const [update] = useUpdateModelPlanMutation();
 
   const handleFormSubmit = (formikValues: StatusFormProps) => {
-    update({
-      variables: {
-        id: modelID,
-        changes: {
-          status: formikValues.status
-        }
-      }
-    })
-      .then(response => {
-        if (!response?.errors) {
-          history.push(`/models/${modelID}/task-list/`);
+    if (formikValues.status) {
+      update({
+        variables: {
+          id: modelID,
+          changes: {
+            status: formikValues.status
+          }
         }
       })
-      .catch(errors => {
-        formikRef?.current?.setErrors(errors);
-      });
+        .then(response => {
+          if (!response?.errors) {
+            showMessageOnNextPage(
+              modelPlanTaskListT('statusUpdateSuccess', {
+                status: statusConfig.options[formikValues.status as ModelStatus]
+              })
+            );
+            history.push(`/models/${modelID}/task-list/`);
+          }
+        })
+        .catch(errors => {
+          formikRef?.current?.setErrors(errors);
+        });
+    }
   };
 
   const initialValues: StatusFormProps = {
-    status: status ?? undefined
+    status: modelStatus ?? status ?? undefined
```

Comment:
This does not seem right. What are you trying to do here?

Reply:
Setting the default to the query param, then the query return, then string